### PR TITLE
Make the project structure closer to a template than an example project

### LINKED ISSUE #3: Alias build peculiarities into a custom `massa-sc-script` NPM package
Currently, we have:

https://github.com/massalabs/massa-sc-template/blob/477e853ec69000ff32ae6fa4c0890fe88294ec76/package.json#L9-L20

I suggest something like:

```json
 "scripts": { 
   "build": "massa-sc-scripts build", 
   "start": "massa-sc-scripts start",
   "test": "massa-sc-scripts test",
   "clean": "massa-sc-scripts clean",
 }, 
```

It will allow user to simply update their SDK by updating `massa-sc-scripts` version in dependencies!

### LINKED ISSUE #5: Remove `include_base64.js` build hack
If you use `node massa_tools/include_base64.js assembly/create_sc.ts` on the following file, the base64 encoding of `get_string.wasm` will be included which is not what we want!
```js
import { create_sc, include_base64 } from "./massa";

export function main(_args: string): i32 {
    // Create smart contract "get_string"
    // const bytes = include_base64('./build/get_string.wasm');
    // create_sc(bytes);
    return 0;
}
```

### PR DESCRIPTION
**Pull request by yvan-sraka**
- Fix #3: Alias build peculiarities into a custom `massa-sc-script` NPM package
- Fix #5: Remove `include_base64.js` build hack

### CODE REVIEW

**Comment by qdrn** [comment]
That looks neeeeat.

However when I run `yarn build` I get the following:
```sh
➜  ~/dev/massa-sc-template (base) (fix-3-and-5) ✗ yarn build                   
yarn run v1.22.17
$ massa-sc-scripts build
/bin/sh: massa-sc-scripts: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Did I miss something ?

**Review by qdrn — COMMENTED** [review]
Don't we need `massa-sc-scripts` to be a fil in the bin directory for this to work ?

Sorry, I'm yarn/npm illiterate.

**Comment by yvan-sraka** [comment]
> Don't we need `massa-sc-scripts` to be a fill in the bin directory for this to work?
> 
> Sorry, I'm yarn/npm illiterate.

Yes, you right! I fixed that locally, I just have to update the `massa-sc-scirpts` NPM package

**Comment by qdrn** [comment]
Ok cool! Sorry I only made a comment and not a review the first time!

File: README.md [hunk 1]
```diff
@@ -25,30 +24,26 @@ yarn install
...
-- `replaceIncludes`: generate a `src/createSC.m.ts`. We want to replace the `include_base64('path.wasm')` with the real binary contained in `path.wasm` before compiling `createSC`.
-- `buildSC`: compile `src/createSC.m.ts`
-- `cleanSC`: remove temp file `src/createSC.m.ts`
-- `createSC`: run `replaceIncludes`, `buildSC` and `cleanSC`
-- `buildAll`: run `helloworld` and `createSC`
-- `exec`: run `buildAll` and then execute the smart contract on a mock with the `massa-sc-test` binary. Look at the mock repository [here](https://github.com/massalabs/massa-sc-tester) for more information about mocking the network.
-- `buildTest`: Build `mytest.ts`
-- `execTest`: Exec in the mocked network `mytest.wasm`
+-   `build`: generate a `.wasm` file in a `build/` directory `src/smart-contract.ts`. This is the hello world smart contract example. If you look at this command line, you can see that we use the `asc` binary, this is the AssemblyScript compiler. Here is [the documentation](https://www.assemblyscript.org/introduction.html) if you want to know more.
```
Comment (qdrn) [inline README.md:33]:
```suggestion
-   `build`: generate a `.wasm` file in a `build/` directory for `src/smart-contract.ts`. This is the hello world smart contract example. If you look at this command line, you can see that we use the `asc` binary, this is the AssemblyScript compiler. Here is [the documentation](https://www.assemblyscript.org/introduction.html) if you want to know more.
```

**Review by adrien-zinger — APPROVED** [review]
The best thing about that is that we can define in an hidden script something to automatically load the correct tester depending on the user configuration! Rendez-vous in massa-sc-script for that. :+1:

Good enough to me

**Comment by qdrn** [comment]
> The best thing about that is that we can define in an hidden script something to automatically load the correct tester depending on the user configuration! Rendez-vous in massa-sc-script for that. 👍

Yes it's quite cool and reduce a lot the complexity and trouble of editing the `package.json` file. What do you imagine for the tester ?

My only concern is regarding the development of multiple smart contracts that interact with each other's. But I think it belongs to a more general question regarding tooling and testing.

**Comment by yvan-sraka** [comment]
Furthermore, I wonder about change the `include_base64(path.wasm)` utility function by a more abstract `bundle(path.wasm)` or `include(path.wasm)` one that let change encoding behind using what's available in a given `massa-sc-std` version, what do you think about it?

**Comment by yvan-sraka** [comment]
> My only concern is regarding the development of multiple smart contracts that interact with each other's. But I think it belongs to a more general question regarding tooling and testing.

What I would suggest is something like `massa-sc-script build src/path.ts -o build/path.wasm` what do you think about it?

**Comment by yvan-sraka** [comment]
For reference on how it was implemented in `react-scripts` (original inspiration):
- [`package.json`](https://github.com/facebook/create-react-app/blob/main/packages/react-scripts/package.json#L27) 
- [`bin/react-scripts.js`](https://github.com/facebook/create-react-app/blob/main/packages/react-scripts/bin/react-scripts.js)

**Comment by qdrn** [comment]
> Furthermore, I wonder about change the `include_base64(path.wasm)` utility function by a more abstract `bundle(path.wasm)` or `include(path.wasm)` one that let change encoding behind using what's available in a given `massa-sc-std` version, what do you think about it?

Yes, why not.

**Review by qdrn — APPROVED** [review]

**Comment by yvan-sraka** [comment]
> What I meant is that this template more or less enforce to create a smart-contract in the following way:
> 
> - one file for smart-contract logic (`smart-contract.ts`)
> 
> - one file to create the smart-contract on the chain (`main.ts`)

That's how it will work after this PR!

> There may be cases where you want to have multiple smart-contracts (multiple `smart-contract_1.ts`, `smart-contract_2.ts`, ...) and upload them using only one smart-contract (`main.ts`). For instance because you need to set multiple rights on the different SC that depend on each SC adresse. In that case we have to fall back to the big old command that is not really convenient. Or maybe I'm missing something ?

Yes, what I mean was to complete `massa-sc-scripts` with a simplified build command that take optional file arguments but keeping in short like `massa-sc-scripts build smart-src/contract_1.ts smart-src/contract_2.ts` would use a default `build` output path that could be customized with an `-o` parameter?!

**Comment by qdrn** [comment]
> would use a default `build` output path that could be customized with an `-o` parameter?!

Ok I see what you mean now! Looks good to me!

**Comment by qdrn** [comment]
> > Don't we need `massa-sc-scripts` to be a fill in the bin directory for this to work?
> > Sorry, I'm yarn/npm illiterate.
> 
> Yes, you right! I fixed that locally, I just have to update the `massa-sc-scripts` NPM package

@yvan-sraka if you have everything setup locally do you mind pushing it so we can update the examples etc...

**Comment by yvan-sraka** [comment]
> @yvan-sraka if you have everything setup locally do you mind pushing it so we can update the examples etc...

Yes, I just update it fixing https://github.com/massalabs/massa-sc-scripts/issues/1#event-6237733697 and following your suggestion of a more flexible `build` command!

**Comment by qdrn** [comment]
I believe there is still a problem on `massa-sc-scripts` part:

```bash
➜  ~/Doc/b/sc-template (base) (fix-3-and-5) yarn install                      127
yarn install v1.22.17
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "massa-sc-scripts > react-dev-utils > fork-ts-checker-webpack-plugin@6.5.0" has unmet peer dependency "typescript@>= 2.7".
warning "massa-sc-scripts > react-dev-utils > fork-ts-checker-webpack-plugin@6.5.0" has unmet peer dependency "webpack@>= 4".
warning " > visitor-as@0.6.0" has incorrect peer dependency "assemblyscript@^0.18.31".
[4/4] 🔨  Building fresh packages...
✨  Done in 1.13s.
➜  ~/Doc/b/sc-template (base) (fix-3-and-5) yarn build  
yarn run v1.22.17
$ massa-sc-scripts build
✨  Done in 0.30s.
➜  ~/Doc/b/sc-template (base) (fix-3-and-5) ls build      
ls: build: No such file or directory
```

**Comment by yvan-sraka** [comment]
> I believe there is still a problem on `massa-sc-scripts` part

Can you confirm that it is solved? What do you think about discussing this issue in the matching PR https://github.com/massalabs/massa-sc-scripts/pull/2?!

**Review by Eitu33 — APPROVED** [review]

**Comment by qdrn** [comment]
Working now! Congratz @Eitu33 @yvan-sraka !

**Pull request merged**